### PR TITLE
fix: Default location prefix would use bucket name

### PR DESCRIPTION
**Pull request by jmgtan**
Issue #517 

## Description of changes:

If no location `databaseName` is provided, the default location prefix would be based on the medallion bucket name

**Checklist**

 

* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/) (`fix: `, `feat: `, `docs: `, ...)

### CODE REVIEW

**Comment by lmouhib** [comment]
GTM, I'll approve once its rebased, not to approve twice

**Comment by lmouhib** [comment]
@jmgtan can you rebase and I will approve

File: framework/src/governance/lib/data-lake-catalog.ts
Comment (vgkowski) [inline framework/src/governance/lib/data-lake-catalog.ts:73]: `const bronzeLocationPrefix = props.databaseName || extractedBronzeBucketName`

Should be like this because we never go into the first condition of `bronzeDatabaseName`

File: framework/test/unit/governance/data-lake-catalog.test.ts
Comment (vgkowski) [inline framework/test/unit/governance/data-lake-catalog.test.ts:217]: Should test the database name contains `exampledb`

**Review by vgkowski — CHANGES_REQUESTED** [review]

File: framework/test/unit/governance/data-lake-catalog.test.ts
Comment (jmgtan) [reply framework/test/unit/governance/data-lake-catalog.test.ts:217]: Fixed

File: framework/src/governance/lib/data-lake-catalog.ts
Comment (jmgtan) [reply framework/src/governance/lib/data-lake-catalog.ts]: Fixed

**Review by vgkowski — APPROVED** [review]

**Pull request merged**
